### PR TITLE
add support for bind mount recursive

### PR DESCRIPTION
**Pull request by ndeloof**
add support for bind mount recursive settings (https://docs.docker.com/engine/storage/bind-mounts/#recursive-mounts)
see https://github.com/docker/compose/issues/11792

### CODE REVIEW

File: schema/compose-spec.json [hunk 1]
```diff
@@ -419,7 +419,8 @@
                     "properties": {
                       "propagation": {"type": "string"},
                       "create_host_path": {"type": ["boolean", "string"]},
-                      "selinux": {"type": "string", "enum": ["z", "Z"]}
+                      "recursive": {"type": "string", "enum": ["z", "Z"]},
+                      "selinux": {"type": "string", "enum": ["enabled", "disabled", "writable", "readonly"]}
```
Comment (glours) [inline schema/compose-spec.json:423]: Inversion?
```suggestion
                      "selinux": {"type": "string", "enum": ["z", "Z"]},
                      "recursive": {"type": "string", "enum": ["enabled", "disabled", "writable", "readonly"]}
```

**Review by jhrotko — APPROVED** [review]
LGTM

File: schema/compose-spec.json (hunk 1 above)
Comment (ndeloof) [reply schema/compose-spec.json:423]: indeed. I'm trying to reduce my caffeine addiction, but this seems to come with side effects

**Review by glours — APPROVED** [review]
LGTM

**Pull request merged**